lxc/config.go

```diff
@@ -101,9 +104,20 @@ func (c *configCmd) run(config *lxd.Config, args []string) error {
 	switch args[0] {
 
 	case "unset":
-		if len(args) < 3 {
+		if len(args) < 2 {
 			return errArgs
 		}
+
+		if len(args) == 2 {
+			key := args[1]
+			c, err := lxd.NewClient(config, "")
+			if err != nil {
+				return err
+			}
+			_, err = c.SetServerConfig(key, "")
+			return err
+		}
+
 		return doSet(config, append(args, ""))
```

Comment:
A minor nitpick (I know I've done this myself), but I think we shouldn't do this (pass the append() result directly into a function). The semantics of append are like that of realloc(), i.e. it can move args to somewhere else. I think what this means is that args after this call is invalid. Although we don't use it for the rest of this function, it is still a code smell.

I introduced this pattern and I'm sorry, but we should probably not continue doing it. I'm fine with either: 1. you can fix this and I'll clean up my own mess or 2. you can promise to fix all of them and I can merge this now.